transport/conn.go

```diff
@@ -323,8 +342,15 @@ func (c *Conn) Query(s Statement, pagingState frame.Bytes) (QueryResult, error)
 func (c *Conn) sendRequest(req frame.Request, compress, tracing bool) (frame.Response, error) {
 	h := make(responseHandler)
 
-	streamID, err := c.r.setHandler(h)
+	c.mu.Lock()
```

Comment:
Conn shall not manage the connReader maybe the mutex should be migrated to conn.